hoot-core/src/main/cpp/hoot/core/visitors/RemoveUnknownVisitor.cpp

```diff
@@ -33,11 +33,13 @@
 namespace hoot
 {
 
+HOOT_FACTORY_REGISTER(ConstElementVisitor, RemoveUnknownVisitor)
 HOOT_FACTORY_REGISTER(ConstElementVisitor, RemoveUnknown1Visitor)
+HOOT_FACTORY_REGISTER(ConstElementVisitor, RemoveUnknown2Visitor)
 
-void RemoveUnknown1Visitor::visit(const boost::shared_ptr<Element>& e)
+void RemoveUnknownVisitor::visit(const boost::shared_ptr<Element>& e)
 {
-  if (e->getStatus() == Status::Unknown1)
+  if (e->getStatus() == _status)
```

Comment:
This is really the meat of the change.  The base class uses a variable to check against the status and each of the derived classes sets that status to a different value.